### PR TITLE
Fixed bug with 180(-180) meridian.

### PR DESCRIPTION
**Pull request by vng**
Just noticed, let's merge this fixes first!

https://github.com/mapbox/cheap-ruler-cpp/pull/12
https://github.com/mapbox/cheap-ruler-cpp/pull/13

### CODE REVIEW

**Review by tmpsantos — APPROVED** [review]

**Comment by mourner** [comment]
I merged the equivalent in https://github.com/mapbox/cheap-ruler/pull/42, let's fill some gaps here and merge too 👍

File: include/mapbox/cheap_ruler.hpp [hunk 1]
```diff
@@ -347,21 +343,30 @@ class CheapRuler {
...
     static point interpolate(point a, point b, double t) {
-        double dx = b.x - a.x;
+        double dx = bounds(b.x - a.x);
         double dy = b.y - a.y;
 
         return point(a.x + dx * t, a.y + dy * t);
     }
 
 private:
+    static double bounds(double lon)
```
Comment (mourner) [inline include/mapbox/cheap_ruler.hpp:361]: let's rename to `wrap` for clarity

File: include/mapbox/cheap_ruler.hpp [hunk 2]
```diff
@@ -347,21 +343,30 @@ class CheapRuler {
...
+    {
+        if (lon < -180.0)
+            return 360.0 + lon;
+        else if (lon > 180.0)
+            return lon - 360.0;
```
Comment (mourner) [inline include/mapbox/cheap_ruler.hpp:366]: Let's do two `while` loops instead to cover cases with more than one wrap.

File: test/cheap_ruler.cpp [hunk 3]
```diff
@@ -39,6 +39,13 @@ TEST_F(CheapRulerTest, distanceInMiles) {
     assertErr(d / d2, 1.609344, 1e-12);
 }
 
+TEST_F(CheapRulerTest, distanceFiji) {
+    cr::CheapRuler ruler(-16.7);
+    auto const d = ruler.distance({ -179.9854832, -16.7813260 }, { 179.8430685, -16.6728974 });
+
+    assertErr(d, 21.8710, 1.0E-3);
+}
+
```
Comment (mourner) [inline test/cheap_ruler.cpp:48]: I've added a bunch of tests for all affected cases in JS — would be nice to cover them equivalently here too.

**Comment by vng** [comment]
@mourner Let's start with merging #12 and #13.

File: include/mapbox/cheap_ruler.hpp [hunk 4]
```diff
@@ -347,21 +343,30 @@ class CheapRuler {
     //
     // Returns true if the given point is inside in the given bounding box, otherwise false.
     //
-    bool insideBBox(point p, box bbox) {
+    static bool insideBBox(point p, box bbox) {
         return p.x >= bbox.min.x &&
                p.x <= bbox.max.x &&
                p.y >= bbox.min.y &&
                p.y <= bbox.max.y;
```
Comment (mourner) [inline include/mapbox/cheap_ruler.hpp:350]: Needs wrapping too.

File: include/mapbox/cheap_ruler.hpp [hunk 5]
```diff
@@ -225,10 +228,7 @@ class CheapRuler {
                 }
             }
 
-            dx = (p.x - x) * kx;
-            dy = (p.y - y) * ky;
-
-            auto sqDist = dx * dx + dy * dy;
+            auto sqDist = squareDistance(p, {x, y});
```
Comment (mourner) [inline include/mapbox/cheap_ruler.hpp:231]: Would this be an added overhead because we're creating a new point object here? Not sure

**Review by mourner — CHANGES_REQUESTED** [review]

File: include/mapbox/cheap_ruler.hpp (hunk 5 above)
Comment (vng) [reply include/mapbox/cheap_ruler.hpp:231]: Zero overhead here ..

**Comment by mourner** [comment]
@vng isn't this PR equivalent to #12?

**Comment by vng** [comment]
Yes, and I insist to merge them first :)

**Pull request closed without merging**